form.py

```
@@ -218,20 +218,16 @@ def to_form_url(string):
         return string
     if string.endswith("viewform"):
         return string.removesuffix("viewform") + "formResponse"
-
-    # If it's an internet shortcut
-    try:
-        parser = ConfigParser()
-        parser.read(string)
-        url = parser["InternetShortcut"]["URL"]
-    except:
-        pass
-    else:
-        # Allow exceptions from this to propagate
-        return to_form_url(url)
-
     raise ValueError(f"String cannot be converted into form link: {string}")
 
+def url_from_shortcut(shortcut):
+    """
+    Return the URL from an internet shortcut.
+    """
+    parser = ConfigParser()
+    parser.read(shortcut)
```

This could be replaced with:

```py
with open(shortcut) as file:
    parser.read_file(file)
```

This makes sure that the file exists (because *ConfigParser.read* silently ignores missing files). It also notes more explicitly that there are two errors to catch in this part (*FileNotFoundError* and *configparser.Error*).

---

Changed in 3026a664df974b3b4c7f42154aafd5ded8c5930f.